jina/clients/helper.py

```diff
@@ -81,3 +83,29 @@ def callback_exec(
         _safe_callback(on_done, continue_on_error, logger)(response)
     if on_always:
         _safe_callback(on_always, continue_on_error, logger)(response)
+
+
+def callback_exec_on_error(
+    on_error: Callable,
+    exception: Exception,
+    logger: JinaLogger,
+    response: Optional = None,
+) -> None:
+    """Execute the on_error callback with the response, Use when anerror outside the responde status was thrown.
+    :param on_error: the on_error callback
+    :param exception: the exception with was thrown and led to the call of on_error
+    :param logger: a logger instance
+    :param response: the response
+    """
+
+    if len(signature(on_error).parameters) == 1:
+        on_error_wrap = on_error
+        warnings.warn(
+            "on_error callback taking only the response parameters is deprecated. Please add one parameter "
+            "to handle additional optional Exception as well",
+            DeprecationWarning,
+        )
+    else:
+        on_error_wrap = lambda resp: on_error(resp, exception)
```

Comment:
I would like this to be done only once? Can we wrap the `on_error` method right when the `Client.post` is done? Like this we will do this only once for all the requests and not once per request?

What I mean, override the `on_error` by the wrap` thing that adds the extra `kwargs`